packages/sd-jwt/src/action-handler.ts

```diff
@@ -0,0 +1,233 @@
+import { Jwt, SDJwt } from '@sd-jwt/core'
+import { SDJwtVcInstance } from '@sd-jwt/sd-jwt-vc'
+import { Signer, Verifier, KbVerifier, JwtPayload } from '@sd-jwt/types'
+import { IAgentPlugin } from '@veramo/core-types'
+import { schema } from './plugin.schema'
+import {
+  SdJWTImplementation,
+  ICreateSdJwtVcArgs,
+  ICreateSdJwtVcResult,
+  ICreateSdJwtVcPresentationArgs,
+  ICreateSdJwtVcPresentationResult,
+  IRequiredContext,
+  ISDJwtPlugin,
+  IVerifySdJwtVcArgs,
+  IVerifySdJwtVcResult,
+  IVerifySdJwtVcPresentationArgs,
+  IVerifySdJwtVcPresentationResult,
+  Claims,
+} from './types.js'
+import { mapIdentifierKeysToDocWithJwkSupport } from '@sphereon/ssi-sdk-ext.did-utils'
+import { encodeJoseBlob } from '@veramo/utils'
+
+/**
+ * SD-JWT plugin for Veramo
+ */
+export class SDJwtPlugin implements IAgentPlugin {
+  readonly schema = schema.ISDJwtPlugin
+
+  constructor(private algorithms: SdJWTImplementation) {}
+
+  // map the methods your plugin is declaring to their implementation
+  readonly methods: ISDJwtPlugin = {
+    createSdJwtVc: this.createSdJwtVc.bind(this),
+    createSdJwtVcPresentation: this.createSdJwtVcPresentation.bind(this),
+    verifySdJwtVc: this.verifySdJwtVc.bind(this),
+    verifySdJwtVcPresentation: this.verifySdJwtVcPresentation.bind(this),
+  }
+
+  /**
+   * Create a signed SD-JWT credential.
+   * @param args - Arguments necessary for the creation of a SD-JWT credential.
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   * @returns A signed SD-JWT credential.
+   */
+  async createSdJwtVc(args: ICreateSdJwtVcArgs, context: IRequiredContext): Promise<ICreateSdJwtVcResult> {
+    const issuer = args.credentialPayload.iss
+    if (!issuer) {
+      throw new Error('credential.issuer must not be empty')
+    }
+    if (issuer.split('#').length === 1) {
```

Comment:
this function should work even if just an issuer DID is provided and then it can check for managed keys for the DID and see if any appropriate ones can be used. of course, a specific key can also be specified

Reply:
I removed the constraint